processing/src/main/java/io/druid/segment/incremental/IncrementalIndex.java

```diff
@@ -924,69 +1009,86 @@ long getTimestamp()
       return timestamp;
     }
 
-    String[][] getDims()
+    int[][] getDims()
     {
       return dims;
     }
 
     @Override
-    public int compareTo(TimeAndDims rhs)
+    public String toString()
+    {
+      return "TimeAndDims{" +
+             "timestamp=" + new DateTime(timestamp) +
+             ", dims=" + Lists.transform(
+          Arrays.asList(dims), new Function<int[], Object>()
+          {
+            @Override
+            public Object apply(@Nullable int[] input)
+            {
+              if (input == null || input.length == 0) {
+                return Arrays.asList("null");
+              }
+              return Arrays.asList(input);
+            }
+          }
+      ) + '}';
+    }
+  }
+
+  Comparator<TimeAndDims> dimsComparator()
+  {
+    return new TimeAndDimsComp();
```

Comment:
TimeAndDimsComp can be a singleton, you can just have a static final INSTANCE in there